modules/core/src/shaderlib/shadow/shadow.js

```diff
@@ -118,13 +119,40 @@ function _getViewportCenterPosition({viewport, center}) {
   return new Matrix4(viewport.viewProjectionMatrix).invert().transformVector4(center);
 }
 
+function _getViewProjectionMatrix({viewport, shadowMatrix}) {
+  const viewMatrix = shadowMatrix.clone().translate(new Vector3(viewport.center).negate());
+
+  const topLeft = viewport.projectPosition(viewport.unproject([0, 0]));
```

Comment:
Perf: make a utility that performs unprojection to the common space from:

https://github.com/uber/deck.gl/blob/master/modules/core/src/viewports/viewport.js#L144

Just skip the `unprojectPosition` part.